platform/darwin/src/NSBundle+MGLAdditions.m

```diff
@@ -6,12 +6,19 @@ @implementation NSBundle (MGLAdditions)
 
 + (instancetype)mgl_frameworkBundle {
     NSBundle *bundle = [self bundleForClass:[MGLAccountManager class]];
-    if (![bundle.infoDictionary[@"CFBundlePackageType"] isEqualToString:@"FMWK"] && !bundle.mgl_resourcesDirectory) {
+
+    if (![bundle.infoDictionary[@"CFBundlePackageType"] isEqualToString:@"FMWK"]) {
         // For static frameworks, the bundle is the containing application
-        // bundle but the resources are still in the framework bundle.
-        bundle = [NSBundle bundleWithPath:[bundle.privateFrameworksPath
-                                           stringByAppendingPathComponent:@"Mapbox.framework"]];
+        // bundle but the resources are in Mapbox.bundle.
+        NSString *bundlePath = [bundle pathForResource:@"Mapbox" ofType:@"bundle"];
+        if (bundlePath) {
+            bundle = [self bundleWithPath:bundlePath];
+        } else {
+            [NSException raise:@"Bundle not found" format:
+             @"The Mapbox framework bundle could not be found in this app. If using the Mapbox iOS SDK as a static framework, make sure that Mapbox.bundle is copied into the root of the app bundle."];
```

Comment:
I was going to suggest something really meta – that we should make this exception localizable like all the others – but then I realized doing so would preempt the exception with infinite recursion! 😆